2011-03-11  Andrew Stubbs  <ams@codesourcery.com>
	    Michael Hope  <michael.hope@linaro.org>
	    Matthias Klose  <doko@ubuntu.org>

	gcc/
	* function. (thread_prologue_and_epilogue_insns): Initialize
	prologue_seq.
	* opts.c (decode_options): Move flag_shrink_wrap to -O1.

=== modified file 'gcc/function.c'
--- old/gcc/function.c	2011-03-10 14:55:10 +0000
+++ new/gcc/function.c	2011-03-11 10:35:05 +0000
@@ -5252,6 +5252,8 @@
   edge_iterator ei;
   bitmap_head bb_flags;
 
+  prologue_seq = NULL_RTX;
+
   df_analyze ();
 
   rtl_profile_for_bb (ENTRY_BLOCK_PTR);

=== modified file 'gcc/opts.c'
--- old/gcc/opts.c	2011-02-08 10:51:58 +0000
+++ new/gcc/opts.c	2011-03-11 10:35:05 +0000
@@ -877,6 +877,7 @@
   flag_tree_copy_prop = opt1;
   flag_tree_sink = opt1;
   flag_tree_ch = opt1;
+  flag_shrink_wrap = opt1;
 
   /* -O2 optimizations.  */
   opt2 = (optimize >= 2);
@@ -909,7 +910,6 @@
   flag_ipa_cp = opt2;
   flag_ipa_sra = opt2;
   flag_ee = opt2;
-  flag_shrink_wrap = opt2;
 
   /* Track fields in field-sensitive alias analysis.  */
   set_param_value ("max-fields-for-field-sensitive",

